pantheon-bundle/src/main/java/com/redhat/pantheon/model/module/ModuleLocale.java

```diff
@@ -25,6 +28,61 @@ default ModuleVersion getOrCreateVersion(String name) {
         return getOrCreateChild(name, ModuleVersion.class);
     }
 
+    /**
+     * Archives the latest released version. Renames the version to the next available archive name.
+     * @return The newly archived {@link ModuleVersion}
+     */
+    default ModuleVersion archiveReleasedVersion() {
+        if(released().get() == null) {
+            // nothing to do
+            return null;
+        }
+
+        // Rename 'released' to the next sequence number
+        String archivedVersionName = generateNextArchiveVersionName();
+        try {
+            return SlingModels.rename(released().get(), archivedVersionName, ModuleVersion.class);
+        } catch (RepositoryException e) {
+            throw new RuntimeException("Problem archiving released version to '" + archivedVersionName + "'", e);
+        }
+    }
+
+    default ModuleVersion releaseDraftVersion() {
+        if(draft().get() == null) {
+            // nothing to do
+            return null;
+        }
+
+        // Rename 'draft' to 'released'
+        try {
+            // TODO externalize this magic string
+            return SlingModels.rename(draft().get(), "released", ModuleVersion.class);
```

Comment:
I'm surprised you don't call `archiveReleasedVersion` defensively prior to executing the rename.

Reply:
You made me think here... :smiley: 
I guess the approach was to have really atomic operations which could be put together in different ways. The defensive call you mention is being done in the servlet itself.